config/routes.rb

```diff
@@ -41,6 +41,11 @@
   get "/register", to: "users#new"
   post "/users", to: "users#create"
   get "/profile", to: "users#show"
+
+  get "/profile/edit", to: "users#edit"
+  patch "/profile", to: "users#update"
+  get "/profile/password/edit", to: "passwords#edit"
+  patch "/profile/password", to: "passwords#update"
```

Comment:
Nice call on creating a `passwords_controller` to handle requests to the password resource.
A password is definitely a nested resource of a profile.